bot/controllers/joke.js

```diff
@@ -0,0 +1,40 @@
+const BaseController = require('../baseController.js');
+const Command = require('../baseCommand.js');
+const util = require('apex-util');
+const request = require('request');
+
+class JokeController extends BaseController {
+  constructor(message) {
+    super(message);
+
+    this.commands = [
+      new Command(
+        '!joke',
+        '!joke',
+        'Tells a joke',
+        'Tells a random geeky joke. Reenter command to get a new joke.',
+        this.jokeAction.bind(this),
+      ),
+    ];
+  }
+
+  // Makes a request to joke API and outputs the joke
+  jokeAction() {
+    const { message } = this;
+
+    if (message) {
+      const options = {
+        url: 'https://geek-jokes.sameerkumar.website/api',
+        method: 'GET',
+      };
+
+      request(options, (err, response, body) => {
+        if (!err && response.statusCode === 200) {
+          util.log(body);
```

Comment:
Only logging the error will result in the user not getting the expected feedback and they will end up sitting and waiting to see if anything is created. A better solution would be to notify the user that the error occurred as well as the dev logs.

Reply:
On second review, you are not logging the error, but only logging in the event of success. The user will not see this functionality.